infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesObjectUtil.java

```diff
@@ -253,4 +257,9 @@ public static boolean shouldCreateInCheNamespace(HasMetadata k8sObject) {
         .getOrDefault(CREATE_IN_CHE_INSTALLATION_NAMESPACE, FALSE.toString())
         .equals(TRUE.toString());
   }
+
+  /** Checks the provided name is a valid kubernetes config map key name */
+  public static boolean isValidConfigMapKeyName(String configMapKeyName) {
+    return VALID_CONFIG_MAP_KEY_NAME_PATTERN.matcher(configMapKeyName).matches();
```

Comment:
what about name max length limitation?

Reply:
good question. I'll check

Reply:
https://kubernetes.io/docs/concepts/configuration/configmap/ I didn't found any notion about length limitation. DO you think it makes sense to limit it, let's say to 63 characters?

Reply:
I would not make any artificial limitation here. Kubernetes will tell when someone hits it :)